Storybook not working

👤:

```
git clone https://github.com/ManageIQ/react-ui-components.git
cd react-ui-components

yarn
yarn run storybook
```

💻:

```
ERROR in ./scripts/storybook/config.js
Module not found: Error: Can't resolve '../../dist/vendor' in '/home/himdel/storybook-github/react-ui-components/scripts/storybook'
 @ ./scripts/storybook/config.js 7:0-28
 @ multi ./node_modules/@storybook/react/dist/server/config/polyfills.js ./node_modules/@storybook/react/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./scripts/storybook/config.js

ERROR in ./scripts/storybook/config.js
Module not found: Error: Can't resolve '../../dist/vendor.css' in '/home/himdel/storybook-github/react-ui-components/scripts/storybook'
 @ ./scripts/storybook/config.js 9:0-32
 @ multi ./node_modules/@storybook/react/dist/server/config/polyfills.js ./node_modules/@storybook/react/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./scripts/storybook/config.js
```

👤: Ok... there's a `vendor` task, still makes relative sense..

```
yarn run vendor
```

💻:

```
ERROR in ../node_modules/css-loader!../node_modules/resolve-url-loader!../node_modules/sass-loader/lib/loader.js!../src/vendor.scss
Module build failed: 
  padding: $jumbotron-padding;
          ^
      Undefined variable: "$jumbotron-padding".
      in /home/himdel/storybook-github/react-ui-components/node_modules/patternfly-react/dist/sass/_card.scss (line 12, column 12)
 @ ../src/vendor.scss 2:14-162
 @ multi ../src/vendor.scss ../src/vendor.js

ERROR in ../src/vendor.scss
Module build failed: ModuleBuildError: Module build failed: 
  padding: $jumbotron-padding;
          ^
      Undefined variable: "$jumbotron-padding".
      in /home/himdel/storybook-github/react-ui-components/node_modules/patternfly-react/dist/sass/_card.scss (line 12, column 12)
    at runLoaders (/home/himdel/storybook-github/react-ui-components/node_modules/webpack/lib/NormalModule.js:195:19)
    at /home/himdel/storybook-github/react-ui-components/node_modules/loader-runner/lib/LoaderRunner.js:364:11
    at /home/himdel/storybook-github/react-ui-components/node_modules/loader-runner/lib/LoaderRunner.js:230:18
    at context.callback (/home/himdel/storybook-github/react-ui-components/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Object.asyncSassJobQueue.push [as callback] (/home/himdel/storybook-github/react-ui-components/node_modules/sass-loader/lib/loader.js:55:13)
    at Object.done [as callback] (/home/himdel/storybook-github/react-ui-components/node_modules/neo-async/async.js:7974:18)
    at options.error (/home/himdel/storybook-github/react-ui-components/node_modules/node-sass/lib/index.js:294:32)
 @ ../src/vendor.scss
 @ multi ../src/vendor.scss ../src/vendor.js
```

👤: ¯\\_(ツ)_/¯ ... ... ok, there's also a `build-storybook` task <small>(but why does it just call itself?)</small> .. let's try that then.

```
yarn run build-storybook
```

💻:

```
Error: => Create a storybook config file in "./.storybook/config.js".
    at exports.default (/home/himdel/storybook-github/react-ui-components/node_modules/@storybook/react/dist/server/config.js:30:15)
    at buildStatic (/home/himdel/storybook-github/react-ui-components/node_modules/@storybook/core/dist/server/build-static.js:77:16)
    at Object.<anonymous> (/home/himdel/storybook-github/react-ui-components/node_modules/@storybook/react/dist/server/build.js:23:25)
```

👤: 🤦

---

BTW, @karelhala re automating this, looks like `yarn` without cache takes about a minute, 20 seconds with cache and 2 s per refresh .. so any tool that does this could concievably be run on demand (assuming storybook build doesn't take significant time (EDIT: 40 seconds without `.out` and 30 seconds with, on the patternfly-react repo)) .. but, if `yarn ; yarn run build-storybook` doesn't Just Work, the tool needs to have some per-repo config that says how to actually get to the place where you can build storybook.

(But, alas, automating this depends on it working in the first place :).)

--- a/.gitignore
+++ b/.gitignore
@@ -42,6 +42,7 @@ dist/*
 *.iws
 *.iml
 
+.out
 coverage
 yarn.lock
 package-lock.json

--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
     "test": "jest",
     "doc": "react-docgen",
     "storybook": "start-storybook -c scripts/storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "build-storybook": "build-storybook -c scripts/storybook -o .out"
   },
   "engines": {
     "node": ">= 6.0.0",

--- a/src/vendor.scss
+++ b/src/vendor.scss
@@ -7,4 +7,4 @@
 /**
   Patternfly React Specific Sass Extensions
 */
-@import '~patternfly-react/dist/sass/patternfly-react';
+@import '~patternfly-react/dist/css/patternfly-react.css';

